bench/benchmarks/geojson_rendering.js

```diff
@@ -0,0 +1,127 @@
+
+var Evented = require('../../js/util/evented');
+var util = require('../../js/util/util');
+var formatNumber = require('../format_number');
+
+module.exports = function(options) {
+  var evented = util.extend({}, Evented);
+
+  var totalFPS = 0;
+  var countRuns = 0;
+
+  var runner = function(zoom) {
+    countRuns++;
+    var map = options.createMap({
+      width: 512,
+      height: 512,
+      zoom: zoom
+    });
+
+    runTest(map, function(fps) {
+      totalFPS+=fps;
+      evented.fire('log', {message: 'zoom '+zoom+' - '+formatNumber(fps)+' fps'});
+      map.remove();
+      if (zoom < 22) {
+        runner(zoom+1);
+      }
+      else {
+        evented.fire('end', {message: 'average across zooms - '+formatNumber(totalFPS/countRuns)});
+      }
+    });
+  }
+
+  runner(0);
+
+  return evented;
+}
+
+var featureCollection = {
+  type: 'FeatureCollection',
+  features: []
+}
+
+for (var idx=0; idx<45; idx+=.15) {
+  featureCollection.features.push({
+    type: 'Feature',
+    properties: {},
+    geometry: {
+      coordinates: [[
+        [0-idx, 0],
+        [0, 0-idx],
+        [0+idx, 0],
+        [0, 0+idx],
+        [0-idx, 0]
+      ]],
+      type: 'Polygon'
+    }
+  });
+}
+
+function runTest(map, cb) {
+  var numFrames = 0;
+  var running = true;
+  var startTime;
+
+  var countFrames = function() {
+    numFrames++;
+    if (running) {
+      requestAnimationFrame(countFrames);
+    }
+  }
+
+  var setData = function(idx) {
+    if (idx <  500) {
+      map.getSource('geojson').setData(featureCollection);
+      setTimeout(function() {
+        setData(idx+1);
+      }, 0);
+    }
+    else {
+      var durration = performance.now() - startTime;
+      var fps = 1000 / (durration / numFrames);
+      running = false;
+      cb(fps);
+    }
+  }
+
+  map.on('load', function() {
+
+    map.addSource('geojson', {
+      data: featureCollection,
+      type: 'geojson'
+    });
+
+    map.addLayer({
+      'source': 'geojson',
+      'id': 'geojson-polygon-fill'
+      type: 'fill',
```

Comment:
There is a syntax error here (no `,` after `'id': 'geojson-polygon-fill'`).

My apologies, I broke the linter in https://github.com/mapbox/mapbox-gl-js/pull/2314 and fixed in #2347. Can you rebase on `master` and fix the lint errors?